### PR TITLE
MNW-226 Add SetPasswordForm

### PR DESCRIPTION
**Pull request by loosefer**
image

### CODE REVIEW

File: packages/frontend/src/components/accounts/SetPassword/lib/validation.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const MIN_PASS_LEN = 8;
+const MAX_PASS_LEN = 20;
```
Comment (NotEternal) [inline packages/frontend/src/components/accounts/SetPassword/lib/validation.ts:2]: Is this a completed limitation? Somebody (for example me) would be uncomfortable with such max password length )

File: packages/frontend/src/components/accounts/SetPassword/ui/ComplexityBlock/lib/complexity.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    if (!value) {
+        return 'none';
+    }
+
+    const week = value.match(regExpWeak);
+    const average = value.match(regExpMedium);
+    const strong = value.match(regExpStrong);
+
+    if (week && average && strong) {
```
Comment (NotEternal) [inline packages/frontend/src/components/accounts/SetPassword/ui/ComplexityBlock/lib/complexity.ts:33]: I reckon we should take into account the pass length here. For example now this is a strong password: "**a1@**". But it's not )

File: packages/frontend/src/components/accounts/SetPassword/ui/ComplexityBlock/ui.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+`;
+
+export const LevelWrapper = styled.div`
+    display: flex;;
+`;
+
+export const Level = styled.div<{ level: number }>`
```
Comment (NotEternal) [inline packages/frontend/src/components/accounts/SetPassword/ui/ComplexityBlock/ui.ts:18]: Maybe add some media rules here? So text won't break.

pass

File: packages/frontend/src/components/accounts/SetupSeedPhrase.js [hunk 4]
```diff
@@ -274,6 +280,22 @@ class SetupSeedPhrase extends Component {
...
                         />
+                        <Route
+                            path={'/setup-seed-phrase/:accountId/set-encryption'}
+                            render={() => (
+                                <Container className='small-centered border'>
+                                    <form
+                                        onSubmit={this.handleSubmitPasswordStep}
+                                        autoComplete='off'
+                                    >
+                                        <h1><Translate id='setupPasswordProtection.pageTitle'/></h1>
```
[the remark this answers is not in the document]
Comment (loosefer) [reply packages/frontend/src/components/accounts/SetupSeedPhrase.js:291]: I suggest not to change anything while, let's keep the same style. When we will ready change and refactor ui, we will remove all shitcode globally

File: packages/frontend/src/components/accounts/SetPassword/ui/ComplexityBlock/lib/complexity.ts (hunk 2 above)
Comment (loosefer) [reply packages/frontend/src/components/accounts/SetPassword/ui/ComplexityBlock/lib/complexity.ts:33]: True

File: packages/frontend/src/components/accounts/ComplexityBlock/ui.ts [hunk 5]
```diff
@@ -12,6 +12,7 @@ export const Content = styled.div`
 
 export const LevelWrapper = styled.div`
     display: flex;;
```
Comment (NotEternal) [inline packages/frontend/src/components/accounts/ComplexityBlock/ui.ts:14]: extra semicolon: `display: flex;;`

File: packages/frontend/src/components/accounts/SetupSeedPhrase.js [hunk 6]
```diff
@@ -211,6 +213,10 @@ class SetupSeedPhrase extends Component {
         e.preventDefault();
     }
 
+    handleSubmitPasswordStep = (password) => {
```
Comment (NotEternal) [inline packages/frontend/src/components/accounts/SetupSeedPhrase.js:216]: Take into account the use of `preventDefault()` so form won't reload.

File: packages/frontend/src/components/accounts/SetPassword/ui/Input/ui.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        background: #ffffff;
+        
+        &:focus {
+            box-shadow: 0px 0px 0px 2px rgba(242, 0, 0, 0.2);
+            border: 1px solid #DC3D43;
+        }
+    `}
+`;
+
+export const IconClickArea = styled.div`
```
Comment (NotEternal) [inline packages/frontend/src/components/accounts/SetPassword/ui/Input/ui.ts:25]: We will remember about addition of some background under the icon

bg

**Review by NotEternal — APPROVED** [review]

File: packages/frontend/src/components/accounts/SetPassword/ui/Input/ui.ts (hunk 7 above)
Comment (loosefer) [reply packages/frontend/src/components/accounts/SetPassword/ui/Input/ui.ts:26]: Fixed with padding-right on input
image

**Pull request merged**